internal/entitycache/cache.go

```diff
@@ -592,7 +600,7 @@ func (iter *Iter) addEntity(e *mongodoc.Entity) bool {
 	entities := &iter.cache.entities
 	entities.mu.Lock()
 	defer entities.mu.Unlock()
-	if _, ok := entities.entities[*e.CanonicalURL()]; ok {
+	if _, ok := entities.entities[*e.CanonicalURLs()[0]]; ok {
```

Comment:
@mhilton nothing special, it's just that any key from CanonicalURLs can be used to test if it's fetched or being fectechd as we store all of them.